cub/cub/device/device_reduce.cuh

```diff
@@ -643,7 +676,7 @@ public:
             typename OutputIteratorT,
             typename NumItemsT,
             typename EnvT = ::cuda::std::execution::env<>>
-  CUB_RUNTIME_FUNCTION static cudaError_t
+  [[nodiscard]] CUB_RUNTIME_FUNCTION static cudaError_t
```

Comment:
Suggestion: there is a feature in `nodiscard` that allows us to print an error message when the result is discarded. Maybe we should use that here and explain why you are supposed to check the error code?

Reply:
it's CUDA suggested behavior globally. I am not sure if we can sum it up on a single step. I am open to change my mind though. Let's take it on a separate issue if you think it's critical